extendeddatacrossword_test.go

```diff
@@ -344,3 +345,143 @@ func createTestEds(codec Codec, shareSize int) *ExtendedDataSquare {
+				nil, nil, nil, nil,
+				nil, nil, nil, nil,
+				nil, nil, nil, nil,
+			},
+			corruptedInd: 0,
+		},
+		{
+			// disturbs the order of shares in the first column, erases the rest of the eds
+			name:          "columns with unordered shares",
+			wantErr:       true, // repair should error out during root construction
+			corruptedAxis: Col,
+			coords: [][]uint{
+				{0, 0},
+				{0, 1},
+				{0, 2},
+				{0, 3},
+				{1, 0},
+				{1, 1},
+				{1, 2},
+				{1, 3},
+				{2, 1},
+				{2, 2},
+				{2, 3},
+				{3, 1},
+				{3, 2},
+				{3, 3},
+			},
+			values: [][]byte{
+				three, nil, nil, nil,
+				one, nil, nil, nil,
+				nil, nil, nil,
+				nil, nil, nil,
+			},
+			corruptedInd: 0,
+		},
+	}
+
+	for codecName, codec := range codecs {
+		t.Run(codecName, func(t *testing.T) {
+			// create a DA header
+			eds := createTestEdsWithNMT(t, codec, shareSize, namespaceSize, 1, 2, 3, 4)
+			assert.NotNil(t, eds)
+			dAHeaderRoots, err := eds.getRowRoots()
+			assert.NoError(t, err)
+
+			dAHeaderCols, err := eds.getColRoots()
+			assert.NoError(t, err)
+			for _, test := range tests {
+				t.Run(test.name, func(t *testing.T) {
+					// create an eds with the given shares
+					corruptEds := createTestEdsWithNMT(t, codec, shareSize, namespaceSize, sharesValue...)
+					assert.NotNil(t, eds)
+					// corrupt it by setting the values at the given coordinates
+					for i, coords := range test.coords {
+						x := coords[0]
+						y := coords[1]
+						corruptEds.setCell(x, y, test.values[i])
+					}
+
+					err = corruptEds.Repair(dAHeaderRoots, dAHeaderCols)
+					assert.Equal(t, err != nil, test.wantErr)
+					if test.wantErr {
+						var byzErr *ErrByzantineData
+						assert.ErrorAs(t, err, &byzErr)
+						errors.As(err, &byzErr)
+						assert.Equal(t, byzErr.Axis, test.corruptedAxis)
+						assert.Equal(t, byzErr.Index, test.corruptedInd)
+					}
+				})
+			}
+		})
+	}
+}
+
+// createTestEdsWithNMT creates an extended data square with the given shares and namespace size.
+// Shares are placed in row-major order.
+// The first namespaceSize bytes of each share are treated as its namespace.
+// Roots of the extended data square are computed using namespace merkle trees.
+func createTestEdsWithNMT(t *testing.T, codec Codec, shareSize, namespaceSize int, sharesValue ...int) *ExtendedDataSquare {
+	// the first namespaceSize bytes of each share are the namespace
+	assert.True(t, shareSize > namespaceSize)
+
+	// create shares of shareSize bytes
+	shares := make([][]byte, len(sharesValue))
+	for i, shareValue := range sharesValue {
+		shares[i] = bytes.Repeat([]byte{byte(shareValue)}, shareSize)
+	}
+	edsWidth := 4            // number of shares per row/column in the extended data square
+	odsWidth := edsWidth / 2 // number of shares per row/column in the original data square
+
+	eds, err := ComputeExtendedDataSquare(shares, codec, newConstructor(uint64(odsWidth), nmt.NamespaceIDSize(namespaceSize)))
+	if err != nil {
+		panic(err)
+	}
```

Comment:
[optional] this can be replaced with

```suggestion
	require.NoError(t, err)
```

Reply:
Sure, done https://github.com/celestiaorg/rsmt2d/pull/242/commits/2dcfb41c7bd30da38fafad1ec303a62bc9b6dfc9.